MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -700,156 +663,74 @@ public String getPortraitFileName() {
+        return (getPrimaryRole() == role) || (getSecondaryRole() == role);
     }
 
     /**
      * @return true if the person has a primary or secondary combat role
      */
     public boolean hasCombatRole() {
-        return hasPrimaryCombatRole() || hasSecondaryCombatRole();
-    }
-
-    /**
-     * @return true if the person has a primary combat role
-     */
-    public boolean hasPrimaryCombatRole() {
-        return hasPrimaryRoleWithin(T_MECHWARRIOR, T_SPACE_GUNNER)
-                || hasPrimaryRoleWithin(T_LAM_PILOT, T_VEHICLE_CREW);
-    }
-
-    /**
-     * @return true if the person has a secondary combat role
-     */
-    public boolean hasSecondaryCombatRole() {
-        return hasSecondaryRoleWithin(T_MECHWARRIOR, T_SPACE_GUNNER)
-                || hasSecondaryRoleWithin(T_LAM_PILOT, T_VEHICLE_CREW);
-    }
-
-    /**
-     * @param includeNoRole whether to include T_NONE in the primary check or not
-     * @return true if the person has a primary or secondary support role, except for secondary T_NONE
-     */
-    public boolean hasSupportRole(boolean includeNoRole) {
-        return hasPrimarySupportRole(includeNoRole) || hasSecondarySupportRole();
-    }
-
-    /**
-     * @param includeNoRole whether to include T_NONE in the check check or not
-     * @return true if the person has a primary support role
-     */
-    public boolean hasPrimarySupportRole(boolean includeNoRole) {
-        return hasPrimaryRoleWithin(T_MECH_TECH, T_ADMIN_HR) || (includeNoRole && hasPrimaryRole(T_NONE));
+        return getPrimaryRole().isCombat() || getSecondaryRole().isCombat();
     }
 
     /**
-     * @return true if the person has a secondary support role. Note that T_NONE is NOT a support
-     * role if it is a secondary role
+     * @param excludeUnmarketable whether to exclude the unmarketable roles from the comparison
+     * @return true if the person has a primary or secondary support role
      */
-    public boolean hasSecondarySupportRole() {
-        return hasSecondaryRoleWithin(T_MECH_TECH, T_ADMIN_HR);
-    }
-
-    /**
-     * Determines whether a role is considered a combat role. Note that T_LAM_PILOT is a special
-     * placeholder which is not used for either primary or secondary role and will return false.
-     *
-     * @param role A value that can be used for a person's primary or secondary role.
-     * @return     Whether the role is considered a combat role
-     */
-    public static boolean isCombatRole(int role) {
-        return ((role > T_NONE) && (role <= T_NAVIGATOR))
-                || (role == T_VEHICLE_CREW);
-    }
-
-    /**
-     * @param role A value that can be used for a person's primary or secondary role.
-     * @return     Whether the role is considered a support role
-     */
-    public static boolean isSupportRole(int role) {
-        return (role >= T_MECH_TECH) && (role < T_LAM_PILOT);
+    public boolean hasSupportRole(boolean excludeUnmarketable) {
```

Comment:
Key note for review: the boolean has been flipped, so anyone using the former value will be doing so incorrectly.